### PR TITLE
feat: customer note migration

### PR DESCRIPTION
**Pull request by Sinucid**
Migration of customer note page. Styles were migrated almost without changes, and contain some drawbacks

closes to [HRZ-2475](https://spryker.atlassian.net/browse/HRZ-2475)

[HRZ-2475]: https://spryker.atlassian.net/browse/HRZ-2475?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/picking/src/components/customer-note/src/customer-note.def.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { componentDef } from '@spryker-oryx/core';
+
+export const customerNoteComponent = componentDef({
+  name: 'oryx-customer-note',
+  impl: () =>
+    import('./customer-note.component').then((m) => m.CustomerNoteComponent),
+});
```
Comment (tarashaliuk) [inline libs/domain/picking/src/components/customer-note/src/customer-note.def.ts:7]: Should we have a schema here like we have in picking-lists and picking-list-item componemts?

File: libs/domain/picking/src/components/customer-note/src/customer-note.model.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface ConfirmPickingOptions {
+  test?: string;
+}
```
Comment (tarashaliuk) [inline libs/domain/picking/src/components/customer-note/src/customer-note.model.ts]: Can't see the usage of this interface. Looks like leftover

File: libs/domain/picking/src/components/navigate-back/src/navigate-back.component.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  @asyncState()
+  protected link = valueType(this.routerService.previousRoute());
+
+  protected override render(): TemplateResult {
+    const link = this.link || this.fallbackUrl;
+
+    return html`
+      <oryx-button type="text">
+        <a href=${link}>
+          <oryx-icon type="backArrow"></oryx-icon>
```
Comment (tarashaliuk) [inline libs/domain/picking/src/components/navigate-back/src/navigate-back.component.ts]:
```suggestion
          <oryx-icon type=${IconTypes.BackArrow}></oryx-icon>
```

File: libs/domain/picking/src/components/navigate-back/src/navigate-back.component.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  protected routerService = resolve(RouterService);
+
+  @asyncState()
+  protected link = valueType(this.routerService.previousRoute());
+
+  protected override render(): TemplateResult {
+    const link = this.link || this.fallbackUrl;
+
+    return html`
+      <oryx-button type="text">
```
Comment (tarashaliuk) [inline libs/domain/picking/src/components/navigate-back/src/navigate-back.component.ts]:
```suggestion
      <oryx-button type=${ButtonType.Text}>
```

File: libs/domain/picking/src/components/navigate-back/src/navigate-back.def.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { componentDef } from '@spryker-oryx/core';
+
+export const navigateBackComponent = componentDef({
+  name: 'oryx-navigate-back',
+  impl: () =>
+    import('./navigate-back.component').then((m) => m.NavigateBackComponent),
+});
```
Comment (tarashaliuk) [inline libs/domain/picking/src/components/navigate-back/src/navigate-back.def.ts]: Please, add schema

File: libs/domain/picking/src/components/navigate-back/src/navigate-back.component.ts (hunk 4 above)
Comment (tarashaliuk) [inline libs/domain/picking/src/components/navigate-back/src/navigate-back.component.ts]: can we use `oryx-icon-button` instead?

File: libs/domain/picking/src/components/picking-list-item/picking-list-item.component.spec.ts [hunk 6]
```diff
@@ -100,30 +113,58 @@ describe('PickingListCardComponent', () => {
 
       expect(event).toHaveBeenCalled();
     });
+
+    describe('and picking is proceed', () => {
+      beforeEach(async () => {
```
Comment (tarashaliuk) [inline libs/domain/picking/src/components/picking-list-item/picking-list-item.component.spec.ts]:
```suggestion
      beforeEach(() => {
```

File: libs/domain/picking/src/components/picking-list-item/picking-list-item.component.ts [hunk 7]
```diff
@@ -1,68 +1,56 @@
...
-    }
-
     this.dispatchEvent(
       new CustomEvent('oryx.show-note', {
         detail: {
-          note: this.pickingList?.cartNote,
+          note: this.pickingList.cartNote,
         },
       })
     );
```
Comment (tarashaliuk) [inline libs/domain/picking/src/components/picking-list-item/picking-list-item.component.ts:56]: In the previous review, you suggested to not dispatch event if there is nothing to show but now you removed the stuff that you suggested. What is the reason? https://github.com/spryker/oryx/pull/111#discussion_r1130756635

File: libs/domain/picking/src/mixins/picking-list.mixin.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  protected pickingList$: Observable<PickingList>;
+  protected pickingList: PickingList;
+}
+
+export const PickingListMixin = <
+  T extends Type<LitElement & PickingListComponentProperties>
+>(
+  superClass: T
+): Type<PickingListMixinInterface> & T => {
+  class AddressMixinClass extends superClass {
```
Comment (tarashaliuk) [inline libs/domain/picking/src/mixins/picking-list.mixin.ts]: Why it is called `AddressMixinClass`?

**Review by tarashaliuk — CHANGES_REQUESTED** [review]

File: libs/domain/picking/src/components/customer-note/src/customer-note.def.ts (hunk 1 above)
Comment (Sinucid) [reply libs/domain/picking/src/components/customer-note/src/customer-note.def.ts:7]: Schema is required for components but this seems to be page.

File: libs/domain/picking/src/components/navigate-back/src/navigate-back.component.ts (hunk 4 above)
Comment (Sinucid) [reply libs/domain/picking/src/components/navigate-back/src/navigate-back.component.ts]: This is a button with type=text regarding designs

File: libs/domain/picking/src/components/picking-list-item/picking-list-item.component.ts (hunk 7 above)
Comment (Sinucid) [reply libs/domain/picking/src/components/picking-list-item/picking-list-item.component.ts:56]: You added conditional rendering (that checks presence of the cartNote) of the icon-button that dispatches this event. Double check is not necessary

File: libs/domain/picking/src/mixins/picking-list.mixin.ts (hunk 8 above)
Comment (Sinucid) [reply libs/domain/picking/src/mixins/picking-list.mixin.ts]: Name is not important, but we need to provide some name for the class that is returned by mixin due to typescript limitation: https://lit.dev/docs/composition/mixins/#applying-decorators-in-mixins

Comment (tarashaliuk) [reply libs/domain/picking/src/mixins/picking-list.mixin.ts]: Let's name it more general like `MixinClass` because `AddressMixinClass` looks strange a bit

Comment (Sinucid) [reply libs/domain/picking/src/mixins/picking-list.mixin.ts]: Oh, sorry, definitely

**Review by tarashaliuk — APPROVED** [review]

**Pull request merged**